### PR TITLE
86cuewym2/Redux - with theme slice.

### PR DESCRIPTION
**Pull request by nijin-26**
### Summary:
Developers can now use the custom hooks `useAppSelector` and `useAppDispatch` for simplified access to the Redux store. The themeSlice can be utilized for managing and updating the application theme.

### Changes:

1. Redux Store Setup: A Redux store has been configured using @reduxjs/toolkit. The store includes a slice for managing the application theme.

2. ThemeSlice: A themeSlice has been added to handle actions related to theme switching. It includes the ability to toggle between light and dark themes and set a specific theme.

3. Custom Hooks for Selector and Dispatch: Custom hooks have been created to simplify the usage of useSelector and useDispatch throughout the application, providing a cleaner and more organized way to interact with the Redux store.

### CODE REVIEW

File: src/app/features/themeSlice.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+interface ThemeState {
```
Comment (abhijith-biju) [inline src/app/features/themeSlice.ts]: prefixing interfaces with I would be better as that convention followed for other interfaces.

File: src/app/store.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export const store = configureStore({
+    reducer: {
+        theme: themeReducer,
+    },
+});
+
+// Infer the `RootState` and `AppDispatch` types from the store itself
+export type RootState = ReturnType<typeof store.getState>;
+// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
```
Comment (abhijith-biju) [inline src/app/store.ts]: Comment specifies incorrect inferred type. 
Is this comment necessary as more slices can be added to the store in future?

File: src/app/hook.ts
Comment (abhijith-biju) [inline src/app/hook.ts:1]: How about hooks.ts as the file name?

**Review by abhijith-biju — CHANGES_REQUESTED** [review]
Please look into the comments.

**Comment by nijin-26** [comment]
> Please look into the comments.

All of those comments are resolved in the recent commit.

File: src/app/hook.ts
Comment (nijin-26) [reply src/app/hook.ts:1]: Yes. It's better. I have updated it.

**Review by amaljoeqb — APPROVED** [review]

**Review by abhijith-biju — APPROVED** [review]

**Review by seraGeorge — APPROVED** [review]

File: src/app/features/themeSlice.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+interface IThemeState {
+    currentTheme: 'light' | 'dark';
```
Comment (lekshmi615213) [inline src/app/features/themeSlice.ts]: enum ?

Comment (nijin-26) [reply src/app/features/themeSlice.ts]: I have updated it in the recent commit.

**Pull request merged**